### ISSUE TITLE
azure.storage.blob.baseblobservice. BaseBlobService .exists logs error if blob does not exist

### ISSUE THREAD

**Issue by tim-werner** [Azure/azure-storage-python#437]
### Which service(blob, file, queue) does this issue concern?
`azure.storage.blob.baseblobservice.BaseBlobService.exists` 

### What problem was encountered?
When checking if a blob exists in container with `azure.storage.blob. baseblobservice.exists`, an error is logged when the blob does not exist in container. The expected behavior would be that no error is logged when the blob does not exist. 

### Have you found a mitigation/solution?
Hint:
`azure.storage.blob.baseblobservice.BaseBlobService.get_blob_properties` sends a request via `_perform_request` that raises an exception which is logged due to https://github.com/Azure/azure-storage-python/blob/master/azure-storage-common/azure/storage/common/storageclient.py#L348

**Comment by zezha-msft** [comment]
Hi @tim-werner, thanks for reaching out. This is a known issue #399. We are actively looking for a solution. I'll update this issue once I have something to share.

**Comment by timotheeguerin** [comment]
Having the same problem when calling `create_container(fail_on_exists=False)`. Solution for us now is to make azure.storage logging level to `CRITICAL`.

Here also I am not 100% sure why you'd need to log this as you are throwing an exception, feels a bit like an anti-pattern. 

Thanks

**Comment by shikharsg** [comment]
temporary fix that will disable all azure storage logging: do this in your python script
```
from azure.storage.common.storageclient import logger
logger.propagate = False
```

**Comment by timotheeguerin** [comment]
I added the following
```python
logging.getLogger("azure.storage").setLevel(logging.CRITICAL)
```

**Comment by zezha-msft** [comment]
Hi @dharmab, 404 from `create_blob_from_text` is quite unexpected, what does the error message say? Is it because the container is missing?

Thanks.

**Comment by dharmab** [comment]
We noticed this with a script that creates a storage account and storage container and generates and uploads certificates. The container and blobs are created and work just fine, we just get error messages in the output. See the attached log for an example.

[storage.txt](https://github.com/Azure/azure-storage-python/files/1997625/storage.txt)

**Comment by dharmab** [comment]
@zezha-msft - you're right, it's actually triggering on `azure.storage.blob.blockblobservice.BlockBlobService.exists`. I've created a smaller script to reproduce the issue [here](https://gist.github.com/dharmab/504db2328cfc1e424a657384d1b4f1e2).

```
$ ./test.py -s $subscription_id -g $resource_group -n $storage_account
Creating storage container "hmwaguxcpudf"
Checking if aqxrf.txt exists
Client-Request-ID=50c33f98-5859-11e8-ab80-d200252a0601 Retry policy did not allow for a retry: Server-Timestamp=Tue, 15 May 2018 16:02:05 GMT, Server-Request-ID=9b7d5497-c01e-00cf-2666-ec5e2b000000, HTTP status code=404, Exception=The specified blob does not exist.ErrorCode: BlobNotFound.
Uploading aqxrf.txt
```

**Comment by zezha-msft** [comment]
Hi @dharmab, thanks for getting back to me.

I'm working on solving this issue of logging expected http errors, and will update this thread as soon as I have something ready.

**Comment by zezha-msft** [comment]
Published in version 1.3.0.

**Comment by tjcrone** [comment]
This error appears to have resurfaced. Is anyone else getting this error again now when a blob does not exist?
